File: src/types/connection.ts
export type DbType = "sqlite" | "postgres";

export type SslMode = "disable" | "prefer" | "require";

export interface PgConfig {
  host: string;
  port: number;
  user: string;
  password: string;
  database: string;
  ssl_mode: SslMode;
}

export interface ConnectionInfo {
  id: string;
  name: string;
  db_type: DbType;
  group: string | null;
  file_path: string | null;
  pg_config: PgConfig | null;
  last_used_at: string | null;
}

File: src-tauri/src/models/connection.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum DbType {
    Sqlite,
    Postgres,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum SslMode {
    Disable,
    Prefer,
    Require,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PgConfig {
    pub host: String,
    pub port: u16,
    pub user: String,
    pub password: String,
    pub database: String,
    pub ssl_mode: SslMode,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectionInfo {
    pub id: String,
    pub name: String,
    pub db_type: DbType,
    pub group: Option<String>,
    pub file_path: Option<String>,
    pub pg_config: Option<PgConfig>,
    pub last_used_at: Option<String>,
}
